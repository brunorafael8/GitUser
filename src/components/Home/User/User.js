'use strict'
import React, { PureComponent } from 'react'
import styled from 'styled-components';
import {
  createFragmentContainer,
  graphql
} from 'react-relay'
class User extends PureComponent {

  render() {
    const {user} = this.props;
    return(
      <div className="container">
        <UserCard>
          <UserDados>
          <UserAvatar src={user.avatarUrl} alt="" />
          <UserName >{user.name}</UserName>
          <UserLogin >{user.login}</UserLogin>
          <span style={styles.UserBio} >{user.bio}</span>
          <span style={styles.UserCompany} >{user.company}</span>
          <span style={styles.UserLocation} >{user.location}</span>
          <Userlinks href={`mailto:${user.email}`} >{user.email}</Userlinks>
          <Userlinks href={user.websiteUrl} >{user.websiteUrl}</Userlinks>
          <UserOrganizaitons style={styles.organizations} >
            {
              user.organizations.edges.map( ({node}) => <a aria-label={node.login} href={node.url} key={node.id}  ><img src={node.avatarUrl} style={styles.UserOrganizaiton} /></a> )
            }
          </UserOrganizaitons>
          </UserDados>
        </UserCard>
      </div>
    )
  }
}
const UserCard = styled.div`
  display: flex;
  margin-top: 20px;
  flex-direction: column;
  @media (max-width: 500px) {
    flex-direction: row;
  }
`
const UserDados = styled.div`
  display: flex;
  flex-direction: column;
  line-height: 30px;
  padding: 20px;
`
const UserOrganizaitons = styled.div`
  display: flex;

`
const UserAvatar = styled.img`
  width: 229px;
  height: 230px;
  @media (max-width: 500px) {
    width: 110px;
    height: 110px;
  }
`
const UserName = styled.span`
  color: '#24292e';
  font-size: 26px;
  media (max-width: 500px) {
    font-size: 16px;
  }
`
const UserLogin = styled.span`
  color: '#666';
  fontSize: 20px;
  media (max-width: 500px) {
    font-size: 14px;
  }
`
const Userlinks = styled.a`
  color: #0366d6;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  media (max-width: 500px) {
    font-size: 14px;

  }
`
const styles = {
  UserBio: {
    color: '#6a737d',
    fontSize: 14
  },
  UserCompany: {
    color: '#24292e',
    fontSize: 14,
    fontWeight: 600
  },
  UserLocation: {
    color: '#24292e',
    fontSize: 14
  },
  UserOrganizaiton: {
    width: 30,
    height: 30
  }
}
export default createFragmentContainer(
  User,
  graphql`
    fragment User_user on User {
      avatarUrl
      name
      login
      bio
      company
      location
      email
      websiteUrl
			organizations(first: 10){
			  edges {
			    node {
			    	id
			    	avatarUrl
            teamsUrl
            url
            login
			    }
        }
      }
    }
  `,
);
