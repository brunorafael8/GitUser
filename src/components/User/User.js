'use strict'
import React, { PureComponent } from 'react'
import styled from 'styled-components';
import {
  createFragmentContainer,
  graphql
} from 'react-relay'

import UserProfile from '../UserProfile/UserProfile'

class User extends PureComponent {

  render() {
    const {user} = this.props;
    return(
      <div className="container">
        <UserProfile user={user} />
      </div>
    )
  }
}

export default createFragmentContainer(
  User,
  graphql`
    fragment User_user on User {
      avatarUrl
      name
      login
      bio
      company
      location
      email
      websiteUrl
			organizations(first: 10){
			  edges {
			    node {
			    	id
			    	avatarUrl
            teamsUrl
            url
            login
			    }
        }
      }
    }
  `,
);
