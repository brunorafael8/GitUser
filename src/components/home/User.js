'use strict'

import React, { PureComponent } from 'react'

class User extends PureComponent {
  render() {
    return(
      <div>

      </div>
    )
  }
}

export default User
