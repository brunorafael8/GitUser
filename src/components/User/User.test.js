'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import User from './User'

import { QueryRenderer, graphql } from 'react-relay'

import environment from '../../environment'

describe('Test UI UserProfile', () => {
  it('The component UserProfile has the expected UI output', () => {
    const query = graphql`
    query UserQuery {
      user (login: "brunorafael8") {
        ... User_user
      }
    }
    `;
    const tree = renderer.create(
      <QueryRenderer
      environment={environment}
      query={query}
      render={({ error, props }) => {
        if (error) console.error(new Error(error));

        if (props) {
          return <User user={props.user} />
        } else {
          return <div>loading ...</div>
        }
      }}
    />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
