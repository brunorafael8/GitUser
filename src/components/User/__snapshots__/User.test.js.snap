// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UI UserProfile The component UserProfile has the expected UI output 1`] = `
<div>
  Searching for , loading ...
</div>
`;
